crates/sp-domains/src/fraud_proof.rs

```diff
@@ -181,6 +181,86 @@ pub enum VerificationError {
     Oneshot(String),
 }
 
+// TODO: Define rest of the fraud proof fields
+#[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
+pub struct MissingInvalidBundleEntryFraudProof {
+    domain_id: DomainId,
+    bundle_index: u32,
+}
+
+impl MissingInvalidBundleEntryFraudProof {
+    pub fn new(domain_id: DomainId, bundle_index: u32) -> Self {
+        Self {
+            domain_id,
+            bundle_index,
+        }
+    }
+}
+
+// TODO: Define rest of the fraud proof fields
+#[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
+pub struct AdditionalInvalidBundleEntryFraudProof {
+    domain_id: DomainId,
+    bundle_index: u32,
+}
+
+impl AdditionalInvalidBundleEntryFraudProof {
+    pub fn new(domain_id: DomainId, bundle_index: u32) -> Self {
+        Self {
+            domain_id,
+            bundle_index,
+        }
+    }
+}
+
+// TODO: Define rest of the fraud proof fields
+#[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
+pub struct MisclassifiedInvalidBundleFraudProof {
+    domain_id: DomainId,
+    bundle_index: u32,
+}
+
+impl MisclassifiedInvalidBundleFraudProof {
+    pub fn new(domain_id: DomainId, bundle_index: u32) -> Self {
+        Self {
+            domain_id,
+            bundle_index,
+        }
+    }
+}
+
+// TODO: Define rest of the fraud proof fields
+#[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
+pub struct IncorrectSortitionFraudProof {
+    domain_id: DomainId,
+}
+
+impl IncorrectSortitionFraudProof {
+    pub fn new(domain_id: DomainId) -> Self {
+        Self { domain_id }
+    }
+}
+
+/// Fraud proof indicating that `invalid_bundles` field of the receipt is incorrect
+#[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
+pub enum InvalidBundlesFraudProof {
+    MissingInvalidBundleEntry(MissingInvalidBundleEntryFraudProof),
+    AdditionalInvalidBundleEntry(AdditionalInvalidBundleEntryFraudProof),
+    MisclassifiedInvalidBundleEntry(MisclassifiedInvalidBundleFraudProof),
+    IncorrectSortition(IncorrectSortitionFraudProof),
```

Comment:
`InvalidBundle::bundle_index` is something I haven't considered before and it indeed can lead to incorrect sortition but I don't think we really need a fraud proof for this.

We only need to add a check to the consensus chain runtime to ensure for all the ER included in the consensus block the `bundle_index` field of `ER.invalid_bundles` must be strictly increasing, see the following proposal for how this can capture the fraud.